dev/com.ibm.ws.jpa.container.core/src/com/ibm/ws/jpa/AbstractJPAProviderIntegration.java

```diff
@@ -126,68 +51,43 @@ public void moduleStopping(ModuleInfo moduleInfo) {}
-                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
-                    Tr.debug(this, tc, "Detected WebSphereLibertyJtaPlatform, not applying dynamic proxy JtaPlatform.");
-            } else {
-                try {
-                    Class<?> JtaPlatform = loadClass(loader, "org.hibernate.engine.transaction.jta.platform.spi.JtaPlatform");
-                    Object libertyJtaPlatform = Proxy.newProxyInstance(loader, new Class[] { JtaPlatform }, new LibertyJtaPlatform());
-                    props.put("hibernate.transaction.jta.platform", libertyJtaPlatform);
-                } catch (ClassNotFoundException x) {
-                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
-                        Tr.debug(this, tc, "Unable to provide JtaPlatform for Liberty TransactionManager to Hibernate", x);
-                }
-            }
-        }
-
         // Log third party provider name and version info once per provider
-        if (providersUsed.add(providerName))
+        if (providersUsed.add(providerName)) {
             logProviderInfo(providerName, puInfo.getClassLoader());
-    };
+        }
+    }
 
-    /**
-     * @see com.ibm.ws.jpa.JPAProvider#modifyPersistenceUnitProperties(java.lang.String, java.util.Properties)
-     */
     @Override
     public void updatePersistenceUnitProperties(String providerClassName, Properties props) {}
 
-    /**
-     * As of Hibernate 5.2.13+ and 5.3+ built-in knowledge of Liberty's transaction integration was delivered as:
-     * org.hibernate.engine.transaction.jta.platform.internal.WebSphereLibertyJtaPlatform
-     * If this class is available, we do not need to set our dynamic proxy instance.
-     */
-    @FFDCIgnore(ClassNotFoundException.class)
-    private boolean isWebSphereLibertyJtaPlatformAvailable(ClassLoader loader) {
-        try {
-            loadClass(loader, "org.hibernate.engine.transaction.jta.platform.internal.WebSphereLibertyJtaPlatform");
-            return true;
-        } catch (ClassNotFoundException notFound) {
-            return false;
+    @FFDCIgnore(PrivilegedActionException.class)
+    protected static Class<?> loadClass(final ClassLoader cl, final String className) throws ClassNotFoundException {
+        if (System.getSecurityManager() == null) {
+            return cl.loadClass(className);
+        } else {
+            try {
+                return AccessController.doPrivileged(new PrivilegedExceptionAction<Class<?>>() {
+                    @Override
+                    public Class<?> run() throws ClassNotFoundException {
+                        return cl.loadClass(className);
+                    }
+                });
+            } catch (PrivilegedActionException e) {
+                if (e.getCause() instanceof ClassNotFoundException) {
+                    throw (ClassNotFoundException) e.getCause();
+                } else {
+                    throw new RuntimeException(e);
```

Comment:
I wonder if we should be throwing the cause (e.getCause()) in the new RuntimeException (if any) rather than the PrivilegedActionException?